tincr/io/device/xdlrc.tcl

```diff
@@ -718,6 +773,63 @@ proc ::tincr::write_partial_primitive_def { site filename {includeConfigs 0} } {
     close $outfile
 }
 
+## Creates a "VCC.def" primitive definition for use in Ultrascale and later devices.
+#   This site does not actually exist on ultrascale parts, but is needed to create
+#   more accurate device descriptions.
+#
+# @param directory Directory to create the "VCC.def" file in
+proc ::tincr::write_vcc_primitive_def { directory } {
+    
+    if { [file isdirectory $directory] == 0 } {
+        puts "$directory is not a valid directory location"
+        return
+    }
+    
+    set outfile [open [file join $directory "VCC.def"] w]
+    
+    
+    puts $outfile "\t(primitive_def VCC 1 2"
+    puts $outfile "\t\t(pin HARD1 HARD1 output)"
+    puts $outfile "\t\t(element HARD1 1"
+    puts $outfile "\t\t\t(pin HARD1 input)"
+    puts $outfile "\t\t\t(conn HARD1 HARD1 <== HARD1VCC P)"
+    puts $outfile "\t\t)"
+    puts $outfile "\t\t(element HARD1VCC 1 # BEL"
+    puts $outfile "\t\t\t(pin P output)"
+    puts $outfile "\t\t\t(conn HARD1VCC P ==> HARD1 HARD1)"
+    puts $outfile "\t\t)"
+    puts -nonewline $outfile "\t)"
+    close $outfile
```

Comment:
Maybe just check `VCC.def` into Git instead of generating a file every time. File operations are notoriously unreliable, and it's good practice to avoid going out to disk whenever possible. I would put it somewhere under [`cache`](https://github.com/byuccl/tincr/tree/master/cache).